src/web/mjs/connectors/SmackJeeves.mjs

```diff
@@ -1,130 +1,99 @@
-                        {
-                            id: comicURL,
-                            title: manga.title,
-                            language: 'en'
-                        }
-                    ];
-                } else {
-                    chapterList = chapterList.map( element => {
-                        return {
-                            id:  new URL( element.value || comicURL, comicURL ).href,
-                            title: element.label.trim() || element.textContent.trim(),
-                            language: 'en'
-                        };
-                    } );
+    async _getChapters(manga) {
+        const url = new URL('/api'+manga.id, this.url);
+        const data = await this._fetchPOST(url);
 
-                }
-                callback( null, chapterList );
-            } )
-            .catch( error => {
-                console.error( error, manga );
-                callback( error, undefined );
-            } );
+        return data.result.list.map(element => {
+            return {
+                id: element.articleUrl,
+                title: this.he.decode( element.articleTitle ).trim()
+            };
+        });
     }
 
-    /**
-     *
-     */
-    _getPageList( manga, chapter, callback ) {
-        let request = new Request( chapter.id, this.requestOptions );
-        this.fetchDOM( request, 'select.jumpbox option.jumpbox_page' )
-            .then( data => {
-                let pageLinks = data.map( element => this.createConnectorURI( this.getAbsolutePath( element.value, request.url ) ) );
-                callback( null, pageLinks );
-            } )
-            .catch( error => {
-                console.error( error, chapter );
-                callback( error, undefined );
-            } );
+    async _getPages(chapter) {
+        const script = `
+            new Promise((resolve, reject) => {
+                try {
+                    resolve(cmnData.comicData);
+                } catch(error) {
+                    reject(error);
+                }
+            });
+        `;
+        const request = new Request(new URL(chapter.id, this.url), this.requestOptions);
+        const data = await Engine.Request.fetchUI(request, script);
+
+        return data.map(element => {
+            return element.url;
+        });
     }
 
-    /**
-     *
-     */
-    _handleConnectorURI( payload ) {
-        let request = new Request( payload, this.requestOptions );
-        /*
-         * TODO: only perform requests when from download manager
-         * or when from browser for preview and selected chapter matches
-         */
-        return this.fetchDOM( request, 'source#comic_image' )
-            .then( data => super._handleConnectorURI( this.getAbsolutePath( data[0], request.url ) ) );
+    async _fetchPOST( uri) {
+        const request = new Request(new URL(uri.pathname, this.url), {
+            method: 'POST',
+            body: uri.searchParams.toString(),
```

Comment:
Placing the form data into the GET parameters of the URL is misleading, i would suggest to separate parameter concerns (what if you have to fetch an URL that actually has some required GET parameters?):

```javascript
async _fetchPOST(uri, form) {
    fetch(uri, {
        body: form.toString()
    });
}

async foobar() {
    let uri = new URL('http://foo.bar?genre=bla')
    let form = new URLSearchParams();
    form.set('token', 123);
    form.set('page', 123);
    let data = await this._fetchPOST(uri, form);
    // ...
```

Reply:
You're right. Can be misleading. Modified.